### PR TITLE
IDVA6-709 - Redirect bookmarked pages

### PR DESCRIPTION
**Pull request by bwallace-ch**
Link to Jira:
https://companieshouse.atlassian.net/browse/IDVA6-709

Changes:
- Redirection functionality on page controllers
- ReferrerUtils function for page redirection
- Unit tests for controllers and redirect function

### CODE REVIEW

**Comment by ch-code-analysis** [comment]
CI: Security warnings found!

**Comment by ch-code-analysis** [comment]
CI: Security warnings found!

**Comment by ch-code-analysis** [comment]
CI: Security warnings found!

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

File: src/routers/controllers/addPresenterController.ts [hunk 1]
```diff
@@ -2,11 +2,39 @@ import { Request, Response } from "express";
...
 import { getUrlWithCompanyNumber } from "../../lib/utils/urlUtils";
 import { AddPresenterHandler } from "../handlers/yourCompanies/addPresenterHandler";
+import { deleteExtraData, getExtraData } from "../../lib/utils/sessionUtils";
+import { redirectPage } from "../../lib/utils/referrerUtils";
 
 export const addPresenterControllerGet = async (req: Request, res: Response): Promise<void> => {
-    const handler = new AddPresenterHandler();
-    const viewData = await handler.execute(req, constants.GET);
-    res.render(constants.ADD_PRESENTER_PAGE, viewData);
+    const referrer: string | undefined = req.get("Referrer");
```
Comment (bmchugh-ch) [inline src/routers/controllers/addPresenterController.ts:9]: Would it be worth moving this checking session data and referrers logic outside of the controllers themselves, in separate functions or did you try that at all? Meaning the navigation checking code can be the same but independent of the controllers.

**Review by bmchugh-ch — CHANGES_REQUESTED** [review]

**Review by mmodzelewski-ch — DISMISSED** [review]

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Review by bmchugh-ch — APPROVED** [review]

**Pull request merged**